ush/python/pygfs/task/stage_ic.py

```diff
@@ -0,0 +1,112 @@
+
+        Parameters
+        ----------
+        config : Dict[str, Any]
+            Incoming configuration for the task from the environment
+
+        Returns
+        -------
+        None
+        """
+        super().__init__(config)
+
+        self.task_config = AttrDict(**self.task_config)
+
+    @logit(logger)
+    def configure(self, stage_dict: Dict[str, Any]) -> (Dict[str, Any]):
+        """Determine stage settings based on configuration and runtime options.
+
+        Parameters
+        ----------
+        stage_dict : Dict[str, Any]
+            Configuration dictionary
+
+        Returns
+        -------
+        stage_dict : Dict[str, Any]
+            Configuration dictionary updated
+        """
+
+        # Add the os.path.exists function to the dict for yaml parsing
+        stage_dict['path_exists'] = os.path.exists
+
+        # Determine model start date
+        current_cycle_begin = add_to_datetime(self.task_config.current_cycle, -to_timedelta(f"{self.task_config['assim_freq']}H") / 2)
+        current_cycle_end = add_to_datetime(self.task_config.current_cycle, to_timedelta(f"{self.task_config['assim_freq']}H") / 2)
+
+        if self.task_config.DOIAU and self.task_config.MODE == "cycled":
+            model_start_date_current_cycle = current_cycle_begin
+        else:
+            if self.task_config.REPLAY_ICS:
+                model_start_date_current_cycle = current_cycle_end
+            else:
+                model_start_date_current_cycle = self.task_config.current_cycle
+
+        stage_dict['model_start_date_current_cycle'] = model_start_date_current_cycle
+
+        # Determine restart RUN
+        rRUN = self.task_config.RUN
+        if self.task_config.RUN == "gfs":
+            rRUN = "gdas"
+        stage_dict['rRUN'] = rRUN
+
+        # Determine ensemble member settings
+        if self.task_config.NMEM_ENS > 0:  # Ensemble RUN
+            if self.task_config.RUN == "gefs":
+                stage_dict['first_mem'] = 0
+                stage_dict['last_mem'] = self.task_config.NMEM_ENS
+            elif self.task_config.RUN == "enkfgdas":
+                stage_dict['first_mem'] = 1
+                stage_dict['last_mem'] = self.task_config.NMEM_ENS
+        else:  # Deterministic RUN
+            stage_dict['first_mem'] = -1
+            stage_dict['last_mem'] = -1
+
+        return stage_dict
+
+    @logit(logger)
+    def execute_stage(self, stage_dict: Dict[str, Any]) -> None:
+        """Perform local staging of initial condition files.
+
+        Parameters
+        ----------
+        stage_dict : Dict[str, Any]
+            Configuration dictionary
+
+        Returns
+        -------
+        None
+        """
+
+        if not os.path.isdir(stage_dict.ROTDIR):
+            raise FileNotFoundError(f"FATAL ERROR: The ROTDIR ({stage_dict.ROTDIR}) does not exist!")
+
+        # Loop over members
+        for mem in range(stage_dict.first_mem, stage_dict.last_mem + 1):
```

Comment:
Looping over members means that the bias correction and deterministic ICs will be copied over and over.
Is this correct?

Reply:
There are separate jobs based on `RUN`. I see I failed to put that in the main PR comment, I have now added that.

When a cycled experiment is set up there is a `gdasstage_ic` job and an `enkfgdasstage_ic` job. The `enkfgdasstage_ic` job won't invoke the block in `stage.yaml.j2` that copies the abias*/radstat files because that block is wrapped in a check of whether `RUN == "gdas"`. See: https://github.com/KateFriedman-NOAA/global-workflow/blob/feature/issue_2475/parm/stage/stage.yaml.j2#L58

See these two logs on Hera for examples:
```
/scratch1/NCEPDEV/stmp4/Kate.Friedman/comrot/teststage_C96C48_ATM/logs/2021122018/gdasstage_ic.log
/scratch1/NCEPDEV/stmp4/Kate.Friedman/comrot/teststage_C96C48_ATM/logs/2021122018/enkfgdasstage_ic.log
```